text/0000-native-cli.md

```diff
@@ -0,0 +1,103 @@
+- Feature Name: native_cli
+- Start Date: 2020-09-17
+- RFC PR: (leave this empty)
+- Neon Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes streamlining the directory layout and command-line UX of Neon projects, making the design simpler, more intuitive, and easier to learn and adopt.
+
+# Motivation
+[motivation]: #motivation
+
+This RFC proposes three simplifications to the Neon design:
+
+- **Simplified model:** A Neon project is no longer a JavaScript package that _contains_ a native module; a Neon project _is just_ a native module.
+- **Simplified CLI:** Using Neon no longer requires learning a custom CLI tool; Neon projects are operated with the standard `npm` and `cargo` tools.
+- **Simplified layout:** A Neon project no longer contains a mix of JS and Rust subdirectory structures; it's flattened into a standard Cargo project + a `package.json` to treat it as a Node native module.
+
+This design lightens the cognitive burden of learning Neon, making it less intimidating for new users and easier to use for everyone.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Creating a new Neon project
+
+The `npm init neon` command makes it easy to start a new Neon project. You can use `npm init neon my-app` to create a new `my-app` Neon directory, or `npm init neon` to initialize an existing directory as a Neon project.
+
+## Neon project layout
+
+A Neon project looks like a typical Rust project that you might create with `cargo new`, but it also contains a `package.json` manifest with the proper npm scripts to build the project as a Node module with `npm install`. After building, a Neon project directory looks like this:
+
+```
+├── Cargo.toml
+├── README.md
+├── build.rs
+├── index.node
+├── package.json
+└── src
+    └── lib.rs
```

Comment:
Nit, this is missing dot files. I think `.gitignore` is the only one we need by default. Supporting ignore files for other version control systems can be left for follow-ups.

Reply:
Agreed, and from testing it out just now (at least with npm 6), it looks like standard `npm init` doesn't create either `.gitignore` or `.npmignore`.